platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java

```diff
@@ -1058,309 +772,307 @@ private void setIsConsensusTrue(EventImpl event, RoundInfo receivedRoundInfo) {
+            if (stronglySeeP(sp, mm) != null && prx == prsp) {
+                x.setStronglySeeP(mm, stronglySeeP(sp, mm));
+            } else if (stronglySeeP(op, mm) != null && prx == prop) {
+                x.setStronglySeeP(mm, stronglySeeP(op, mm));
+            } else {
+                // the canonical witness by mm that is seen by x thru someone else
+                final EventImpl st = seeThru(x, mm, mm);
+                if (round(st) != prx) { // ignore if the canonical is in the wrong round, or doesn't exist
+                    x.setStronglySeeP(mm, null);
                 } else {
-                    EventImpl st =
-                            seeThru(x, mm, mm); // the canonical witness by mm that is seen by x thru someone else
-                    if (round(st) != prx) { // ignore if the canonical is in the wrong round, or doesn't exist
-                        x.setStronglySeeP(mm, null);
-                    } else {
-                        long weight = 0;
-                        for (int m3 = 0; m3 < numMembers; m3++) {
-                            if (seeThru(x, mm, m3) == st) { // only count intermediates that see the canonical witness
-                                final NodeId nodeId = addressBook.getNodeId(m3);
-                                weight += addressBook.getAddress(nodeId).getWeight();
-                            }
-                        }
-                        if (SUPER_MAJORITY.isSatisfiedBy(weight, totalWeight)) { // strongly see supermajority of
-                            // intermediates
-                            x.setStronglySeeP(mm, st);
-                        } else {
-                            x.setStronglySeeP(mm, null);
+                    long weight = 0;
+                    for (int m3 = 0; m3 < numMembers; m3++) {
+                        if (seeThru(x, mm, m3) == st) { // only count intermediates that see the canonical witness
+                            weight += getWeight(m3);
                         }
                     }
+                    if (Threshold.SUPER_MAJORITY.isSatisfiedBy(weight, totalWeight)) { // strongly see supermajority of
+                        // intermediates
+                        x.setStronglySeeP(mm, st);
+                    } else {
+                        x.setStronglySeeP(mm, null);
+                    }
                 }
             }
-            result = x.getStronglySeeP(m);
         }
-        t = System.nanoTime() - t; // nanoseconds spent doing the dot product
-        consensusMetrics.dotProductTime(t);
-        return result;
+        return x.getStronglySeeP((int) m);
     }
 
     /**
-     * The round-created for event x (first round is 1), or 0 if x is null (function from SWIRLDS-TR-2020-01).
-     * It also stores the round number with x.setRoundCreated().
-     * This result is memoized.
+     * The round-created for event x (first round is 1), or 0 if x is null (function from
```

Comment:
These javadoc comments no longer correspond to the code.   Please update the comments.